sway-core/src/semantic_analysis/ast_node/expression/typed_expression/function_application.rs

```diff
@@ -85,6 +83,8 @@ pub(crate) fn instantiate_function_application(
     {
         cached_fn_ref
     } else {
+        let mut function_decl = TyFunctionDecl::clone(&*function_decl);
```

Comment:
I am getting mixed feelings, between the safety of the qualified method call to avoid cloning the `Arc` by chance, and the familiarity of the `(&*function_decl).clone()`. But safety first! 😄